libpdbg/device.c

```diff
@@ -149,7 +229,28 @@ static int dt_cmp_subnodes(const struct pdbg_target *a, const struct pdbg_target
 		if (*a_end == 0 && *b_end == 0)
 			return (a_num > b_num) - (a_num < b_num);
 	}
+	char a_name_without_num[strlen(a->dn_name) + 1];
+	char b_name_without_num[strlen(b->dn_name) + 1];
+	unsigned int a_only_num = 0, b_only_num = 0;
 
+	/* Create a copy as we are going to chop the number from it 
+	 *and we dont want to update the original string
+	 */
+	strcpy(a_name_without_num, a->dn_name);
+	strcpy(b_name_without_num, b->dn_name);
+
+	/* Collect the numbers into variables to compare */
+	a_only_num = strip_num_at_the_end(a_name_without_num);
+	b_only_num = strip_num_at_the_end(b_name_without_num);
+
+	/* If both of the targets are of the same group, sort using the number at the end
+	 * Ensure that the target has a number at the end 
+	 */
+	if(strcmp(a_name_without_num, b_name_without_num) == 0 &&
```

Comment:
The condition check can be modified as follows to make it more faster.
`if ((a_only_num != 0xFFFF) && (b_only_num != 0xFFFF) && (strcmp(a_name_without_num, b_name_without_num) == 0))` i.e.; interchanging the positions of the conditions checks. Reason, the `strcmp` function is a performance intensive operations and putting it first we make the process slow unnecessarily if one of the number returns false for default value check.